src/librustdoc/clean/mod.rs

```diff
@@ -872,14 +873,33 @@ impl Attributes {
     pub fn links(&self) -> Vec<(String, String)> {
         use html::format::href;
         self.links.iter().filter_map(|&(ref s, did, ref fragment)| {
-            if let Some((mut href, ..)) = href(did) {
-                if let Some(ref fragment) = *fragment {
-                    href.push_str("#");
-                    href.push_str(fragment);
+            match did {
+                Some(did) => {
+                    if let Some((mut href, ..)) = href(did) {
+                        if let Some(ref fragment) = *fragment {
+                            href.push_str("#");
+                            href.push_str(fragment);
+                        }
+                        Some((s.clone(), href))
+                    } else {
+                        None
+                    }
+                }
+                None => {
+                    if let Some(ref fragment) = *fragment {
+                        // This is a primitive so the url is done "by hand".
+                        Some((s.clone(),
+                              format!("https://doc.rust-lang.org/{}std/primitive.{}.html",
```

Comment:
We don't link to the stable std docs anywhere, since libstd itself has a `doc_html_root` set to the nightly docs. I would prefer if you could somehow grab the url root for std to combine with the "hand-made" primitive link, or at least always link to nightly.